src/Data/List.purs

```diff
@@ -860,7 +880,10 @@ foldM f b (a : as) = f b a >>= \b' -> foldM f b' as
 -- |
 -- | Running time: `O(n)`
 mapReverse :: forall a b. (a -> b) -> List a -> List b
-mapReverse f = go Nil
+mapReverse f l = reverse $ map f l
```

Comment:
Using this slower, but definitely stack safe alternative (not saying the other version is stack unsafe, but just trying to eliminate unknowns).